content/docs/command-reference/stage/add.md

```diff
@@ -0,0 +1,432 @@
+is reproduced (see also `dvc gc`).
+
+Relevant notes:
+
+- Typically, scripts to run (or possibly a directory containing the source code)
+  are included among the specified `-d` dependencies. This ensures that when the
+  source code changes, DVC knows that the stage needs to be reproduced. (You can
+  chose whether to do this.)
+
+- `dvc stage add` checks the dependency graph integrity before creating a new
+  stage. For example: two stage cannot specify the same output or overlapping
+  output paths, there should be no cycles, etc.
+
+- DVC does not feed dependency files to the command being run. The program will
+  have to read by itself the files specified with `-d`.
+
+- Entire directories produced by the stage can be tracked as outputs by DVC,
+  which generates a single `.dir` entry in the cache (refer to
+  [Structure of cache directory](/doc/user-guide/project-structure/internal-files#structure-of-the-cache-directory)
+  for more info.)
+
+- [external dependencies](/doc/user-guide/external-dependencies) and
+  [external outputs](/doc/user-guide/managing-external-data) (outside of the
+  <abbr>workspace</abbr>) are also supported (except metrics and plots).
+
+- Outputs are deleted from the workspace before executing the command (including
+  at `dvc repro`) if their paths are found as existing files/directories (unless
+  `--outs-persist` is used). This also means that the stage command needs to
+  recreate any directory structures defined as outputs every time its executed
+  by DVC.
+
+- In some situations, we have previously executed a stage, and later notice that
+  some of the files/directories used by the stage as dependencies, or created as
+  outputs are missing from `dvc.yaml`. It is possible to
+  [add missing dependencies/outputs to an existing stage](/docs/user-guide/how-to/add-deps-or-outs-to-a-stage)
+  without having to execute it again.
+
+- Renaming dependencies or outputs requires a
+  [manual process](/doc/command-reference/move#renaming-stage-outputs) to update
+  `dvc.yaml` and the project's cache accordingly.
+
+### For displaying and comparing data science experiments
+
+[parameters](/doc/command-reference/params) (`-p`/`--params` option) are a
+special type of key/value dependencies. Multiple parameter dependencies can be
+specified from within one or more YAML, JSON, TOML, or Python parameters files
+(e.g. `params.yaml`). This allows tracking experimental hyperparameters easily.
+
+Special types of output files, [metrics](/doc/command-reference/metrics) (`-m`
+and `-M` options) and [plots](/doc/command-reference/plots) (`--plots` and
+`--plots-no-cache` options), are also supported. Metrics and plots files have
+specific formats (JSON, YAML, CSV, or TSV) and allow displaying and comparing
+data science experiments.
+
+### The `command` argument
+
+The `command` sent to `dvc stage add` can be anything your terminal would accept
+and run directly, for example a shell built-in, expression, or binary found in
```

Comment:
I know this comes exactly from the `run` ref. but should we take the opportunity to move this before `### Dependencies and outputs`?